pydantic_ai_slim/pydantic_ai/durable_exec/dbos/_agent.py

```diff
@@ -0,0 +1,668 @@
+        *,
+        name: str | None = None,
+        event_stream_handler: EventStreamHandler[AgentDepsT] | None = None,
+        mcp_step_config: StepConfig | None = None,
+        model_step_config: StepConfig | None = None,
+    ):
+        """Wrap an agent to enable it with DBOS durable workflows, by automatically offloading model requests, tool calls, and MCP server communication to DBOS steps.
+
+        After wrapping, the original agent can still be used as normal outside of the DBOS workflow.
+
+        Args:
+            wrapped: The agent to wrap.
+            name: Optional unique agent name to use as the DBOS configured instance name. If not provided, the agent's `name` will be used.
+            event_stream_handler: Optional event stream handler to use instead of the one set on the wrapped agent.
+            mcp_step_config: The base DBOS step config to use for MCP server steps. If no config is provided, use the default settings of DBOS.
+            model_step_config: The DBOS step config to use for model request steps. If no config is provided, use the default settings of DBOS.
+        """
+        super().__init__(wrapped)
+
+        self._name = name or wrapped.name
+        if self._name is None:
+            raise UserError(
+                "An agent needs to have a unique `name` in order to be used with DBOS. The name will be used to identify the agent's workflows and steps."
+            )
+
+        # Merge the config with the default DBOS config
+        self._mcp_step_config = mcp_step_config or {}
+        self._model_step_config = model_step_config or {}
+
+        if not isinstance(wrapped.model, Model):
+            raise UserError(
+                'An agent needs to have a `model` in order to be used with DBOS, it cannot be set at agent run time.'
+            )
+
+        dbos_model = DBOSModel(
+            wrapped.model,
+            step_name_prefix=self._name,
+            step_config=self._model_step_config,
+            event_stream_handler=event_stream_handler or wrapped.event_stream_handler,
+        )
+        self._model = dbos_model
+
+        dbosagent_name = self._name
+
+        def dbosify_toolset(toolset: AbstractToolset[AgentDepsT]) -> AbstractToolset[AgentDepsT]:
+            # Replace MCPServer with DBOSMCPServer
+            if isinstance(toolset, MCPServer):
+                return DBOSMCPServer(
+                    wrapped=toolset,
+                    step_name_prefix=dbosagent_name,
+                    step_config=self._mcp_step_config,
+                )
+            else:
+                return toolset
+
+        dbos_toolsets = [toolset.visit_and_replace(dbosify_toolset) for toolset in wrapped.toolsets]
+        self._toolsets = dbos_toolsets
+        DBOSConfiguredInstance.__init__(self, self._name)
+
+        # Wrap the `run` method in a DBOS workflow
+        @DBOS.workflow(name=f'{self._name}.run')
+        async def wrapped_run_workflow(
```

Comment:
Please update this (and the other methods that take all of these arguments) with the new `deferred_tool_results` argument added in v1.0.0b1